lib/spack/spack/environment/environment.py

```diff
@@ -912,6 +916,31 @@ def included_config_scopes(self):
             # allow paths to contain spack config/environment variables, etc.
             config_path = substitute_path_variables(config_path)
 
+            # strip file URL prefix, if needed
+            config_path = config_path.replace('file://', '')
+
+            if not os.path.exists(config_path):
```

Comment:
Done

Reply:
Actually it doesn't.  `is_url_format` checks whether the path (string) conforms to the URL format , which includes absolute local path as currently defined.

Consequently, without the check an absolute path (without previously staged remote configuration files) will result in `fetch_remote_configs` unnecessarily being called and `ConfigFileError` being raised because it does not process local directories.

Reply:
Thanks: I completely missed that absolute paths match the regex in `is_url_format`. In that case though: would this potentially try to download `config_path` when it didn't exist (e.g. if you enter the wrong absolute path on your FS, and it is absolute, this function would fail in a confusing manner because it would try to download that file and fail)?

Reply:
> ...would this potentially try to download `config_path` when it didn't exist (e.g. if you enter the wrong absolute path on your FS, and it is absolute, this function would fail in a confusing manner because it would try to download that file and fail)?

Yes to download.  The resulting failure is `spack.util.web.FetchFailureError` and the explanation includes `No such file or directory`.  I added a unit test to cover that case (`test_env_with_included_config_missing_file`).